### DISCUSSION TITLE
Why do some directories not want to show up? And how to avoid wine clutter?

### DISCUSSION THREAD

**Discussion by indeedwatson** [junegunn/fzf#3213] [Q&A]
For example, I have ~/Music and ~/Series

Both directories are symlinked from another drive, yet triggering fzf and typing `mus` immediately yields `~/Music`, but the same is not true for series.

What shows up is unrelated directories and *a lot* of wine prefix subdirectories. In fact wine subdirectories are basically spam when I search for something I can't find, can I set fzf to avoid going into any wine prefix?

**Comment by sweetbbak** [comment]
There is an environment variable that sets the default fzf command. I forget what they use as a command default but this is how I have mine set in a file that I source in my .zshrc and .bashrc. You can set it anywhere so long as you shell can find the environment variable.
```
export FZF_DEFAULT_COMMAND="rg ~ --files --hidden"
```
This uses ripgrep to find files and hidden files in the home dir as the default fzf command.
You can use any command here like "find" or "fd" and then use the options on one of these commands to ignore certain directories, filetypes or files that match a certain regex.
```
export FZF_DEFAULT_COMMAND="fd . ~ -E '.wine'"
```

I would personally just do this manually and write a quick little bash script that uses find or fd to look only in the dirs that I want. ie:
```
mpv $(fd . ~/Music -t f --extension mp4 -e mp3 | fzf --cycle)

# or

fd . ~ -tf -e mp4 -e mp3 -e epub | fzf --preview 'preview-script {}' | xargs mpv
```
but xargs and mpv sometimes dont mix, so I recommend the first one or some variant of that. Depending on your terminal you can write a preview script (or find one like fzf-ueberzug) and use that to preview the files with images. Ueberzug is really the only image viewer that works with fzf now, icat/kitty used to work but not anymore. You could have a case statement that reads the filetype and then decides how to preview it. exiftool is a good script to use to look at audio/video and picture files and will tell you their length and more.

**Comment by indeedwatson** [comment]
Thank you. I'd really like to avoid specific commands.

For the wine prefixes I'd probably have to exclude the `drive_c` and `dosdevices` folders.

For the folders that don't show up, I'm not sure how your suggestions would help with that, because I don't actually know what the issue is, or why it's happening.

I didn't know about exiftool, that sounds very useful!

EDIT: this simple screenshot should encapsulate my problem
![1679620870](https://user-images.githubusercontent.com/10749504/227400309-6db4746e-03db-47e4-b20b-db27374defad.png)

**Reply by sweetbbak** [reply]
```
rg --files --iglob '!dosdevices' --iglob '!drive_c' --iglob '!go/pkg' | fzf
```
This works for me for excluding directories with ripgrep and Im able to search for directories like Music and Videos. The default command environment variable just sets what command fzf uses by default when invoked outside of having stdin piped in to it, so my guess is that Films/ isnt in there in the first place. It also may be best to put that into your zshrc and then restarting your shell to eliminate any issues, the changed environment variable might not even be having any effect.

What happens if you just do "find | fzf" or even "cd ~ && for x in **/*; do echo $x; done | fzf " and then try to type Films in the query?

**Comment by indeedwatson** [comment]
I did put it in .zshrc, the screenshot was for clarity's sake. The first command does ignore the wine directories when I launch it like that, maybe `alt+c` doesn't take into account `FZF_DEFAULT_COMMAND`? Or maybe wrapping single quotes inside double quotes messes it up? I'm out of ideas.

Although `~/Films/` still does not show up, but it does show up with `find | fzf`

**Reply by sweetbbak** [reply]
export FZF_ALT_C_COMMAND="fd -t d -d 1"
export FZF_ALT_C_OPTS="--preview 'tree -C {} | head -200' --height=60%"

alt+c uses a different environment variable lol. Im sure its just an issue with what you have set for one of these variables, otherwise piping stuff to fzf would also not work. Maybe even unsetting them, or resetting them back to the default would help you trouble shoot that. The manual page suggests running the command "$SHELL -c "to see if its posix compliant. ``` "$SHELL" -c find```

```
# Integration with ripgrep
RG_PREFIX="rg --column --line-number --no-heading --color=always --smart-case "
INITIAL_QUERY="foobar"
FZF_DEFAULT_COMMAND="$RG_PREFIX '$INITIAL_QUERY'" \
fzf --bind "change:reload:$RG_PREFIX {q} || true" \
    --ansi --disabled --query "$INITIAL_QUERY"
```
It also has this as a good example

**Comment by indeedwatson** [comment]
I never set those variables.

The first example returns a very limited set of results. I'm unsure what to do with the rest of the information you provide. I tried unsetting them but it doesn't change the behavior.

I also tried setting the alt c command variable to the same command as FZF_DEFAULT_COMMAND, but that doesn't work at all.

EDIT I have now tried replacing all the env variables with simple `fd -L`. `Films` does show up, but if I enable hidden files, then the results are again cluttered by wine. It is so ridiculous in fact, that it finds the correct folder but inside a wine prefix in the `z` volume.
